### DISCUSSION TITLE
Connecting data from multiple components

### DISCUSSION THREAD

**Discussion by michaldev** [django-commons/django-unicorn#785] [Ideas]
Hi. I have one-page site with cart on bottom of the page and catalog in other position on the same page.

I would like to create action for click "Add item" in catalog. After clicking item should append to cart. For this solution I have to pack all from page for one component - its dirty. Can I create 2 other components - cart and catalog, and integrate it?

**Comment by adamghill** [comment]
There has been some discussion around this topic in other posts, but there isn't a real clean way to do what you are asking -- at least not yet.

One way to approach it now would be to have `add item` store information in a database or external cache (i.e. redis or memcache) and have the cart component poll for new items. But, that feels a little hacky.

You could also call call a custom JavaScript method from the `add item` component view (https://www.django-unicorn.com/docs/advanced/#call-javascript-from-view) which in turn uses `Unicorn.call` to call a `catalog` component view method (https://www.django-unicorn.com/docs/actions/#calling-methods). I've never experimented with this, but I think it should work.

It'd be nice to provide a cleaner API to solve this use case in the future, so if you have a suggestion about what might make sense I'd love to hear it.

**Comment by JohananOppongAmoateng** [comment]
@adamghill i think we can convert this into a discussion
